ERCS/erc-7662.md

```diff
@@ -0,0 +1,273 @@
+
+It is RECOMMENDED to add a data to the [ERC-721](./eip-721.md) standard that makes it easy for e.g. NFT Marketplaces to display data about the AI Agent NFT, i.e. Model, which in turn reveals the platform that is used for the agent, e.g. OpenAI in the case of gpt-4-0125-preview or Anthropic in the case of claude-3-opus-20240229. The standard name and description can be used to display the Agent Name and Agent Description. 
+
+## Rationale
+
+This standard provides a unified way to create and parse AI Agent NFTs. 
+
+This standard codifies the necessary parameters of Name, Description, Model, User Prompt, and System Prompt for creating and using AI Agent NFTs. 
+
+It doesn't make practical sense to store the user and system prompts in an existing [ERC-721](./eip-721.md) as the only place to put would be in the token metadata that is open for anyone to access the prompts without owning the NFT. By storing the prompts in a custom Agent struct and restricting access to the prompts to the holder of the NFT.  One way to do this would be through restricing access to the struct info to the holder of the NFT through a custom function, however since that option still exposes the prompt URIs to the public and thus the data inside them, the recommended method is by encrypting the prompts onchain and tying the decryption of the URLs to the holder of the NFT, using onchain services such as Lit Protocol that enable decryption to be tied to contract parameters such as ownerOf(tokenId).
+ 
+
+## Backwards Compatibility
+
+The AI Agents NFT standard introduces additional features and data to the standard [ERC-721](./eip-721.md) protocol, aimed at addressing the practical requirements of using NFTs to store, trade and use AI Agents. It is designed to be fully backward-compatible with the original [ERC-721](./eip-721.md) standard.  All existing [ERC-721](./eip-721.md) functions (such as transferFrom, approve, and balanceOf) retain their original functionality and interfaces. Our extension does not modify these core behaviors, ensuring that any [ERC-721](./eip-721.md) compliant wallet or service can interact with these tokens without modifications.
+
+### Reference Implementation
+
+This is being currently implemented in a CEO AI product for creating, managing and using AI Agents Onchain through a DApp interface. In this implementation, Lit Protocol is being used to encrypt the prompts using custom EVMContractParameters that only decrypt for the holder of the NFT and using Arweave to store the URLs of this encrypted data. To facilitate that and make DApp handling easier, some parameters were added to Agent and the addEncryptedPrompts function is added that enables adding the encrypted prompt URIs after first minting the NFT (as the tokenId of the NFT is needed for setting the encryption/decryption conditions)
+
+```solidity
+
+// SPDX-License-Identifier: MIT
```

Comment:
Code in the proposal itself must be licensed under `CC0-1.0`. You may put MIT code in the assets directory if you wish.